### ISSUE TITLE
Fatal Judge Shutdown Error

### ISSUE THREAD

**Issue by quantum5** [DMOJ/site#6]
Removing a dead judge should not involve taking down the bridge.

```pytb
Traceback (most recent call last):

  File "/code/site/judge/bridge/judgehandler.py", line 165, in packet
    handler(data)

  File "/code/site/judge/bridge/judgecallback.py", line 85, in on_grading_end
    super(DjangoJudgeHandler, self).on_grading_end(packet)

  File "/code/site/judge/bridge/judgehandler.py", line 187, in on_grading_end
    self._free_self(packet)

  File "/code/site/judge/bridge/judgehandler.py", line 237, in _free_self
    self.server.judges.on_judge_free(self, packet['submission-id'])

  File "/code/site/judge/bridge/judgelist.py", line 57, in on_judge_free
    self._handle_free_judge(judge)

  File "/code/site/judge/bridge/judgelist.py", line 26, in _handle_free_judge
    self.judges.remove(judge)

KeyError: <judge.bridge.judgecallback.DjangoJudgeHandler object at 0x22a9e10>
```

**Comment by quantum5** [comment]
Probably have something to do with two threads freeing the same judge.

Would be fixed in 89ce5d6a2af3a32ac1bebf4950865306c8e1be75 if that is the case.

**Comment by quantum5** [comment]
Closing because never reproduced ever after.
